funsor/contract.py

```diff
@@ -0,0 +1,242 @@
+from __future__ import absolute_import, division, print_function
+
+from collections import OrderedDict
+from six import integer_types
+
+import opt_einsum
+import torch
+
+import funsor.ops as ops
+from funsor.distributions import Gaussian, Delta
+from funsor.interpreter import interpretation, reinterpret
+from funsor.optimizer import Finitary, optimize
+from funsor.sum_product import _partition
+from funsor.terms import Funsor, Number, Reduce, Variable, eager
+from funsor.torch import Tensor
+
+
+ATOMS = (Tensor, Gaussian, Delta, Number, Variable)
```

Comment:
Should we add `Joint` to this list, or are you decomposing that into a sum?

Reply:
Ah I see this is triaged. Would you mind adding a #TODO in the code to help me remember?